Hosted Launch: Run first DAG (#2237)
astronomer/docs: astro/create-first-dag.md

- Create an Astro project. 
- Authenticate and log in to Astro. 
- Create a Deployment. 
- Deploy your DAGs to Astro with the Astro CLI.
- Trigger a run of an example DAG in the Airflow UI. 

## Prerequisites

> 

Reviewer: I don't know about this, it feels like we are projecting because veteran Astro users obviously understand the need of Docker for local testing, but a new user likely won't have any assumption about needing Docker?

> 

Reviewer 2: Good point. I think just because Astro CLI install docs have Docker on there + obviously so many people have Docker problems (and we actively tried to make these steps Docker-less).

I'd keep it, but maybe add a teeny bit of context?

I just don't like it in the bullet points below

> - The [Astro CLI](cli/install-cli.md), but you do not need to install Docker to complete this guide

Reviewer: Should we just go ahead and remove Docker from the CLI install prerequisites? Or list it there as optional if you want to run Airflow locally/ deploy project changes?

Author: Before this PR gets merged, I realized that I haven't been able to test if `astro dev init` works without Docker. There were a couple of messages that got printed during my build and in the descriptions of the core commands (https://docs.astronomer.io/astro/cli/overview#core-commands) that make me a little wary of removing docker if it's that important to what we consider core commands or actions.

Author: I'm going to come back and update this section after I finish with the CLI draft. For right now, I think it makes sense to keep that docker isn't a requirement for this quickstart.

- An Astronomer account
- An Astro Organization and [Workspace](manage-workspaces.md)
- An Astro [cluster](create-cluster.md)

If you're the first person on your team to try Astro, see [Start your Astro trial](astro/trial.md). If your team has an existing Organization on Astro that you are not a member of, see [Add a user to an Organization](astro/add-user#add-a-user-to-an-organization).

:::info

> This guide shows you how to deploy your DAGs to Astro, which does not require Docker. For a standard deployment of a complete Astro project, see [Deploy Code](deploy-code.md).

Reviewer: ```suggestion
This guide shows you how to deploy only your DAG files to Astro. To deploy other project changes, such as Python dependencies, see [Deploy Code](deploy-code.md).
```

- Create an Astro project. 
- Authenticate and log in to Astro. 

> - Create a Deployment. 

Reviewer 2: Change up order here to match new steps?

- An Astronomer account
- An Astro Organization and [Workspace](manage-workspaces.md)

If you're the first person on your team to try Astro, see [Start your Astro trial](astro/trial.md). If your team has an existing Organization on Astro that you are not a member of, see [Add a user to an Organization](astro/add-user#add-a-user-to-an-organization).

## Step 1: Start a trial

Go to [Try Astro](https://www.astronomer.io/try-astro/) to activate your free 14-day trial. To create your Astro user account, you'll need to provide a valid email address and create a password.

## Step 2: Create an Organization and Workspace

After you've created your Astro user account, you're asked to create an Organization and your first Workspace. 

An _Organization_ is the highest management level on Astro. An Organization contains _Workspaces_, which are collections of Deployments, or Airflow environments.

To start your trial, Astronomer recommends using the name of your company as the name of your Organization and naming your first Workspace after your data team or initial business use case with Airflow. You can update these names in the Cloud UI after you finish activating your trial. 

> ## Step 3: Log in to Astro CLI and Cloud UI

Reviewer 2: Should we skip straight to create a Deployment before we log the user in? Feels weird to flip them to CLI.

Re: auth to UI -- keep it to the "create your user account" part?

To run DAGs on the Cloud UI, you need to authenticate with both the Astro CLI and Cloud UI.

1. Run the following command to authenticate to Astro on the CLI:

    ```sh
    astro login
    ```

    After running this command, you are prompted to open your web browser and enter your credentials to the Cloud UI. Then, you are automatically authenticated to the CLI.

2. Go to `https://cloud.astronomer.io`, and select one of the available options to access the Cloud UI.

## Step 4: Create a Deployment

> An Astro _Deployment_ is an instance of Airflow that hosts all core Airflow components, including a webserver, scheduler, and one or more workers. You can have one or more Deployments within a Workspace.

Reviewer 2: ```suggestion
An Astro _Deployment_ is an instance of Apache Airflow that is powered by all core Airflow components, including a webserver, scheduler, and one or more workers. You deploy DAGs to a Deployment, and you can have one or more Deployments within a Workspace.
```

  
    You can leave the other fields at their default values. This creates a basic Deployment on a standard Astronomer-hosted cluster. You can delete the Deployment after you finish testing your example DAG runs. 

4. Click **Create Deployment**.

     A confirmation message appears indicating that the Deployment status is **Creating** until all underlying components in the Deployment healthy. During this time, the Airflow UI is unavailable and you can't deploy code or modify Deployment settings. When the Deployment is ready, the status changes to **Healthy**.
    
    For more information about possible Deployment health statuses, see [Deployment health](deployment-metrics.md#deployment-health). Or, to learn more about how to customize your Deployment settings, see [Configure a Deployment](configure-deployment-resources).

## Step 5: Create an Astro project

An _Astro project_ contains the set of files necessary to run Airflow, including dedicated folders for your DAG files, plugins, and dependencies. All new Astro projects contain two example DAGs. In this tutorial, you'll be deploying these example DAGs to your Deployment on Astro.

1. Open your terminal or IDE

> 2. Create a new folder for your Astro project:

Reviewer 2: Style guide addition!

> 2. Go to `https://cloud.astronomer.io`, and select one of the available options to access the Cloud UI.

Reviewer: Auth docs FAQs are primarily published to catch users that are used to our previous login workflow, they don't mean much to brand new users. I think it's okay to not link those here.

## Step 3: Create a Deployment

> An Astro _Deployment_ is an Astro Runtime environment that is powered by the core components of Apache Airflow, including the Airflow webserver, scheduler, and one or more workers. You can have one or more Deployments within a _Workspace_, which is a collection of users that have access to the same Deployments.

Reviewer 2: ```suggestion
An Astro _Deployment_ is an instance of Apache Airflow that can run DAGs and is powered by core Airflow components, including the Airflow webserver, scheduler, and one or more workers. You can have one or more Deployments within a _Workspace_, which is a collection of users that have access to the same Deployments. In most cases, Astronomer recommends creating a Deployment for each logical set of DAGs that you want to manage. You can create more dedicated Deployments over time that represent development or production environments.
```

Too much? Gimme feedback (and know we haven't defined glossary), but thought a lil more context would be good

Reviewer: Also "Instance of Airflow.. powered by core Airflow components" feels redundant. I like "instance of Airflow that hosts all core Airflow components

When you create a Deployment on Astro, infrastructure is created in your cluster that runs your DAGs with all core Apache Airflow components.

1. In the Cloud UI, select a **Workspace**.

2. On the **Deployments** page, click **+ Deployment**.

> 3. Complete the following fields:

Reviewer 2: Do we need update the **Cluster** thing here? Maybe say something that pushes them to `Standard`? Should we explicitly say assume defaults for everything else?

>     For more information about possible Deployment health statuses, see [Deployment health](deployment-metrics.md#deployment-health). Or, to learn more about how to customize your Deployment settings, see [Configure a Deployment](configure-deployment-resources).

Reviewer 2: ^There's some cluster language up top that I'd remove -- simplify to: `... until all underlying Airflow components are healthy`.

## Step 4: Deploy your DAGs to your Astro Deployment

> You can use DAG-only deploys to quickly update your Astro Deployment.

Reviewer 2: We def shouldn't spend too much time here, but I'd add half a sentence to clarify what we're talking about. At least call it a "feature" that "only deploy the `dags` folder of your Astro project, which allows you to deploy changes more quickly without Docker and without having to restart your Airflow environment."

Something like that

> You can use DAG-only deploys to quickly update your Astro Deployment.

Reviewer 2: I'd also specify here that the DAGs we're deploying are the example DAGs we alluded to at the top. Maybe say that you're free to add your own in addition to that

> You can use DAG-only deploys to quickly update your Astro Deployment without building a Docker image of your entire project. 

Reviewer: ```suggestion
You can use DAG-only deploys to quickly update your Astro Deployment.
```
Not sure we need to mention Docker at this stage, WDYT?

> 1. Run the following command to enable DAG-only code deploys.

Reviewer: ```suggestion
1. Run the following command to enable DAG-only code deploys on your Deployment.
```

```sh
astro deploy
```
1. Run the following command to enable DAG-only code deploys on your Deployment.

This command returns a list of Deployments available in your Workspace and prompts you to pick one.
    ```sh
    astro deployment update --dag-deploy enable
    ```

> After you select a Deployment, the CLI parses your DAGs to ensure that they don't contain basic syntax and import errors. If your code passes the parse, the Astro CLI builds all files in your Astro project directory into a new Docker image and then pushes the image to your Deployment on Astro. 

Reviewer 2: Hm feels like we still want some variety of this sentence even if it's shorter. And DAGs still parse right? That's a feature/value-add we should still have if so!

> 3. Run the following command to finalize the setup and trigger a DAG-only deploy to your Deployment:  

Reviewer 2: ```suggestion
3. Run the following command to deploy your DAGs to Astro: 
```

    This command returns a list of Deployments available in your Workspace and prompts you to confirm where you want to deploy your DAG code.

> ## Step 6: Trigger your DAG on Astro

Reviewer: ```suggestion
## Step 5: Trigger your DAG on Astro
```
Oops forgot this!

> ## Step 5: Trigger your DAG on Astro

Reviewer 2: ```suggestion
## Step 5: Trigger your DAG in Airflow UI
```

Maybe?

Reviewer: Hm I know that we will have another way of doing this "On Astro", but to start talking about the "Airflow UI" at this stage could be confusing. I'd read this and think I have to go somewhere other than Astro to access the Airflow UI

Reviewer 3: I see that the intent in this doc is to have the user go through the Airflow UI. I think it'd be good to change that to the Astro UI in the near future ( maybe give the new DAG page in Astro a couple of weeks to bed in )

> When you deploy your Astro project, all example DAGs are paused and will not start running automatically. To run one of these example DAGs according to its schedule, you must unpause it from the Airflow UI.

Reviewer 2: ```suggestion
Any time that you deploy new DAGs to Astro, they are paused and will not start running automatically. To run one of the example DAGs in your Astro project according to its schedule, you must unpause it from the Airflow UI.
```

5. Pause your DAG by clicking the slider button next to `example-dag-basic`. This prevents your example DAG from running automatically and consuming your Deployment resources.

## Step 6: View your DAG status in the Cloud UI

> Your Airflow UI view allows you to examine your DAGs in detail, while the Cloud UI shows you information about the health of your Deployment including analytics and logs of your DAG runs.

Reviewer 3: I get that the intent is to drive users to the Airflow UI but I think this part now seems a bit confusing given the DAGs page(s) in Astro? Like Astro lets you examine your DAG in detail too..

> Your Airflow UI view allows you to examine your DAGs in detail, while the Cloud UI shows you information about the health of your Deployment including analytics and logs of your DAG runs.

Reviewer: ```suggestion
The Cloud UI shows you information about the health of your Deployment, including analytics and logs for your DAG runs.
```

5. Pause your DAG by clicking the slider button next to `example-dag-basic`. This prevents your example DAG from running automatically and consuming your Deployment resources.

## Step 6: View your DAG status in the Cloud UI

Your Airflow UI view allows you to examine your DAGs in detail, while the Cloud UI shows you information about the health of your Deployment including analytics and logs of your DAG runs.

Go back to your Deployment page in the Cloud UI. Because you ran your example DAG, your Deployment information page now has data about your Deployment and DAG runs. The following example shows an example of what you might find in the **Analytics** view for your Deployment.

![Summary information about your DAG runs in the Analytics tab of a Quickstart Deployment.](/img/docs/first-DAG-data.png)

> ## Step 7: (Optional) Delete your Deployment

Reviewer 2: Debating on whether we include this vs. just include as note in next steps -- will let you make the call @jwitz but wanted to throw it out there

Reviewer: Hm I'd like to make it more apparent first to show that resource management is a core part of the Astro workflow. It's also useful to surface this now because it's not super clear in the product where to do this

## Next Steps

> Now that you've created and run your first DAG on Astro, the next step is to learn how to deploy a complete Astro Project to the Cloud UI.

Reviewer: ```suggestion
Now that you've created and run your first DAG on Astro, the next step is to learn how to deploy a complete Astro Project to the Cloud UI. See:
```

- [Install Docker](https://www.docker.com/products/docker-desktop/) to use the full capabilities of the Astro CLI, including the ability to run Airflow locally and customize your Deployment's Airflow environment. 

> - [Deploy code to Astro](/astro/deploy-code)
> - [Develop a project](/astro/develop-project)

Reviewer: ```suggestion
- [Develop a project](/astro/develop-project)
- [Deploy code to Astro](/astro/deploy-code)
```

> - [Learn how to write your own DAGs](https://docs.astronomer.io/learn/category/dags)

Reviewer: ```suggestion
```

> - [Set up CI/CD](/astro/set-up-ci-cd)

Reviewer: ```suggestion
```

## Next Steps

> Now that you've created and run your first DAG on Astro, the next step is to learn how to deploy a complete Astro Project to the Cloud UI. See:

Reviewer 2: ```suggestion
Now that you've created and run your first DAG on Astro, the next step is to add your own DAGs, build out the rest of your Astro project, and start testing real data. See:
```

Reviewer 2: Yea, the "full project" thing with Docker + Astro CLI is a little confusing -- not sure what answer is but give it another look @jwitz

## Next Steps

Now that you've created and run your first DAG on Astro, the next step is to add your own DAGs, build out the rest of your Astro project, and start testing real data. See:

> - [Install Docker](https://www.docker.com/products/docker-desktop/) to use the full capabilities of the Astro CLI, including the ability to run Airflow locally and customize your Deployment's Airflow environment. 

Reviewer 2: ```suggestion
- [Install Docker](https://www.docker.com/products/docker-desktop/) to use the full capabilities of the Astro CLI, including the ability to run Airflow locally and deploy the rest of your Astro project to Astro, including Python packages.
```

- [Deploy code to Astro](/astro/deploy-code)
- [Develop a project](/astro/develop-project)

## Trial limitations

Astro trials have some limitations that aren't present in the paid product:

> - You can only create Deployments in standard clusters, which are clusters that are shared with other Astro users. 

Reviewer 2: Call it multi-tenant here?

- You can only have up to two Deployments at any given time.
- You can only configure the `default` worker queue, and you can only have a maximum worker count of two.
   
## After your trial

After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms. Any DAGs you deployed will continue to run for an additional 7-day grace period.

After the 7-day grace period, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:

> - Schedule a call with your point of contact from Astronomer.

Reviewer 2: I'd just cut these bullet points to make it shorter -- just say "contact sales" include email address (ask @ElanDWeiner)
